primitives/src/targeting/eval.rs

```diff
@@ -411,6 +722,102 @@ fn eval(input: &Input, output: &mut Output, rule: &Rule) -> Result<Option<Value>
 
             Some(Value::Bool(a.iter().any(|x| b.contains(x))))
         }
+        Function::In(first_rule, second_rule) => {
+            let a = eval(input, output, first_rule)?.ok_or(Error::TypeError)?;
+            let b = eval(input, output, second_rule)?
+                .ok_or(Error::TypeError)?
+                .try_array()?;
+
+            Some(Value::Bool(b.contains(&a)))
+        }
+        Function::NotIn(first_rule, second_rule) => {
+            let a = eval(input, output, first_rule)?.ok_or(Error::TypeError)?;
+            let b = eval(input, output, second_rule)?
+                .ok_or(Error::TypeError)?
+                .try_array()?;
+
+            Some(Value::Bool(!b.contains(&a)))
+        }
+        Function::Between(first_rule, second_rule, third_rule) => {
+            let first_eval = first_rule.eval(input, output)?.ok_or(Error::TypeError)?;
+            let second_eval = second_rule.eval(input, output)?.ok_or(Error::TypeError)?;
+            let third_eval = third_rule.eval(input, output)?.ok_or(Error::TypeError)?;
+
+            let value = BigNum::try_from(first_eval)?;
+            let start = BigNum::try_from(second_eval)?;
+            let end = BigNum::try_from(third_eval)?;
+
+            Some(Value::Bool(value.ge(&start) && value.le(&end)))
```

Comment:
I assumed the second value will always be smaller than the third value

Reply:
even if it's not then 1 of the 2 checks will fail. I.e. if third value (end) is smaller than second (start) then either `value.ge(&strart)` will fail or `value.le(&end)` as `value` can't satisfy both at the same time.